planner/core/find_best_task.go

```diff
@@ -581,6 +581,14 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty) (t task, err
 }
 
 func (ds *DataSource) canConvertToPointGet(candidate *candidatePath) bool {
+	// If the schema contains ExtraPidColID, do not convert to point get.
+	// Because the point get executor can not handle the extra partition ID column now.
+	for _, col := range ds.schema.Columns {
```

Comment:
/LGTM

Only change this when the table is partition table is better for performance.